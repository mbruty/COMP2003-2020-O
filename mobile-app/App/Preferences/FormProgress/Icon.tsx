import React from "react";
import {
  Image,
  ImageSourcePropType,
  StyleSheet,
  Text,
  View,
} from "react-native";

interface Props {
  image: ImageSourcePropType;
  text: string;
}

export const Icon: React.FC<Props> = (props) => {
  return (
    <View style={styles.container} key={`icon ${props.text}`}>
      <Image style={styles.img} source={props.image} />
      <Text allowFontScaling={false} style={styles.txt}>{props.text}</Text>
    </View>
  );
};

const styles = StyleSheet.create({
  container: {
    alignItems: "center",
  },
  img: {
    height: 35,
    width: 35,
  },
  txt: {
    textAlign: "center",
  },
});
