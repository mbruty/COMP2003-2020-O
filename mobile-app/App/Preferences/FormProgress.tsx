import React from "react";
import {
  Dimensions,
  ImageSourcePropType,
  StyleSheet,
  Text,
  TouchableOpacity,
  TouchableOpacityComponent,
  View,
} from "react-native";
import { CONSTANT_STYLES } from "../constants";
import { Icon } from "./FormProgress/Icon";

// ToDo: If this is ever needed any where else, maybe add this to the props?
const items: Array<{
  text: string;
  img: { active: ImageSourcePropType; inactive: ImageSourcePropType };
}> = [
  {
    text: "Create Your \nAccount",
    img: {
      active: require("./FormProgress/account_circle_active.png"),
      inactive: require("./FormProgress/account_circle_inactive.png"),
    },
  },
  {
    text: "Confirm Your \nEmail",
    img: {
      active: require("./FormProgress/email_active.png"),
      inactive: require("./FormProgress/email_inactive.png"),
    },
  },
  {
    text: "Select Your \nPreferences",
    img: {
      active: require("./FormProgress/food_active.png"),
      inactive: require("./FormProgress/food_inactive.png"),
    },
  },
];

interface Props {
  selectedIdx: number;
}
export const FormProgress: React.FC<Props> = (props) => {
  const dimensions = Dimensions.get("window");
  const imageWidth = dimensions.width;
  return (
    <>
      <View style={[styles.container, { width: imageWidth - 25 }]}>
        {items.map((x, index) => (
          <>
            <Icon
              text={x.text}
              image={
                index === props.selectedIdx ? x.img.active : x.img.inactive
              }
            />

            {index < items.length - 1 && <View style={styles.line} />}
          </>
        ))}
      </View>
      <View style={[styles.container, { marginTop: -15 }]}>
        <TouchableOpacity
          style={[
            props.selectedIdx === 0
              ? CONSTANT_STYLES.BG_LIGHT_GREY
              : CONSTANT_STYLES.BG_DARK_GREY,
            styles.btn,
            { marginRight: 150 },
          ]}
        >
          <Text style={[CONSTANT_STYLES.TXT_BASE, styles.txt]}>BACK</Text>
        </TouchableOpacity>
        <TouchableOpacity style={[CONSTANT_STYLES.BG_RED, styles.btn]}>
        <Text style={[CONSTANT_STYLES.TXT_BASE, styles.txt]}>{props.selectedIdx === 2 ? "SUBMIT" : "NEXT"}</Text>
        </TouchableOpacity>
      </View>
    </>
  );
};

const styles = StyleSheet.create({
  container: {
    flex: 1,
    flexDirection: "row",
    justifyContent: "space-between",
    alignSelf: "center",
    padding: 15,
    paddingTop: 25,
  },
  line: {
    height: 2,
    backgroundColor: "#707070",
    width: 50,
    alignSelf: "center",
    marginTop: -35,
    flexShrink: 1,
  },
  btn: {
    paddingLeft: 25,
    paddingRight: 25,
    paddingTop: 10,
    paddingBottom: 10,
    borderRadius: 10,
  },
  txt: {
    fontWeight: "bold",
  },
});
