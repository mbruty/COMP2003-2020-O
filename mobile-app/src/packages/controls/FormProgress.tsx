import React from "react";
import {
  Dimensions,
  ImageSourcePropType,
  StyleSheet,
  Text,
  TouchableOpacity,
  TouchableOpacityComponent,
  View,
} from "react-native";
import { CONSTANT_STYLES } from "../../constants";
import { Icon } from "./Icon";

// ToDo: If this is ever needed any where else, maybe add this to the props?
const items: Array<{ name: string; icon: string }> = [
  { name: "Create Your \nAccount", icon: "account-box" },
  { name: "Confirm Your \nEmail", icon: "email" },
  { name: "Select Your \nPreferences", icon: "food" },
];

interface Props {
  selectedIdx: number;
  allowBack: boolean;
  onSubmit: () => void;
  onBack?: () => void;
  text?: string;
}
const FormProgress: React.FC<Props> = (props) => {
  const dimensions = Dimensions.get("window");
  const imageWidth = dimensions.width;
  const text = props.text || "SUMBIT";
  return (
    <>
      <View
        key={`FormProgress ${props.selectedIdx}`}
        style={[
          styles.container,
          {
            width: imageWidth - 25,
            justifyContent: "space-between",
            alignSelf: "center",
          },
        ]}
      >
        {items.map((x, index) => (
          <>
            <Icon
              text={x.name}
              iconName={x.icon}
              enabled={props.selectedIdx === index}
              index={index}
            />

            {index < items.length - 1 && <View style={styles.line} />}
          </>
        ))}
      </View>
      <View style={[styles.container, { paddingBottom: 25 }]}>
        {props.allowBack && (
          <TouchableOpacity
            style={[
              props.selectedIdx === 0
                ? CONSTANT_STYLES.BG_LIGHT_GREY
                : CONSTANT_STYLES.BG_DARK_GREY,
              styles.btn,
              { left: 25, bottom: 0 },
            ]}
          >
            <Text style={[CONSTANT_STYLES.TXT_BASE, styles.txt]}>BACK</Text>
          </TouchableOpacity>
        )}
        <TouchableOpacity
          style={[CONSTANT_STYLES.BG_RED, styles.btn, { right: 25, bottom: 0 }]}
          onPress={props.onSubmit}
        >
          <Text
            allowFontScaling={false}
            style={[CONSTANT_STYLES.TXT_BASE, styles.txt]}
          >
            {text}
          </Text>
        </TouchableOpacity>
      </View>
    </>
  );
};

const styles = StyleSheet.create({
  container: {
    flex: 1,
    flexDirection: "row",
    padding: 15,
    paddingTop: 25,
  },
  line: {
    height: 2,
    backgroundColor: "#707070",
    width: 50,
    alignSelf: "center",
    marginTop: -35,
    flexShrink: 1,
  },
  btn: {
    paddingLeft: 20,
    paddingRight: 20,
    paddingTop: 8,
    paddingBottom: 8,
    borderRadius: 10,
    position: "absolute",
  },
  txt: {
    fontWeight: "bold",
  },
});
export default FormProgress;
