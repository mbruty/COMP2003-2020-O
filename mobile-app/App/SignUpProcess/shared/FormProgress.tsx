import React from "react";
import {
  Dimensions,
  ImageSourcePropType,
  StyleSheet,
  Text,
  TouchableOpacity,
  TouchableOpacityComponent,
  View,
} from "react-native";
import { CONSTANT_STYLES } from "../../shared/constants";
import { Icon } from "../Preferences/FormProgress/Icon";

// ToDo: If this is ever needed any where else, maybe add this to the props?
const items: Array<{
  text: string;
  img: { active: ImageSourcePropType; inactive: ImageSourcePropType };
}> = [
  {
    text: "Create Your \nAccount",
    img: {
      active: require("../Preferences/FormProgress/account_circle_active.png"),
      inactive: require("../Preferences/FormProgress/account_circle_inactive.png"),
    },
  },
  {
    text: "Confirm Your \nEmail",
    img: {
      active: require("../Preferences/FormProgress/email_active.png"),
      inactive: require("../Preferences/FormProgress/email_inactive.png"),
    },
  },
  {
    text: "Select Your \nPreferences",
    img: {
      active: require("../Preferences/FormProgress/food_active.png"),
      inactive: require("../Preferences/FormProgress/food_inactive.png"),
    },
  },
];

interface Props {
  selectedIdx: number;
  allowBack: boolean;
  onSubmit: () => void;
  onBack?: () => void;
}
export const FormProgress: React.FC<Props> = (props) => {
  const dimensions = Dimensions.get("window");
  const imageWidth = dimensions.width;
  return (
    <>
      <View
        key={`FormProgress ${props.selectedIdx}`}
        style={[
          styles.container,
          {
            width: imageWidth - 25,
            justifyContent: "space-between",
            alignSelf: "center",
          },
        ]}
      >
        {items.map((x, index) => (
          <>
            <Icon
              text={x.text}
              image={
                index === props.selectedIdx ? x.img.active : x.img.inactive
              }
            />

            {index < items.length - 1 && <View style={styles.line} />}
          </>
        ))}
      </View>
      <View style={[styles.container, {paddingBottom: 25}]}>
        {props.allowBack && (
          <TouchableOpacity
            style={[
              props.selectedIdx === 0
                ? CONSTANT_STYLES.BG_LIGHT_GREY
                : CONSTANT_STYLES.BG_DARK_GREY,
              styles.btn,
              { left: 25, bottom: 0 },
            ]}
          >
            <Text style={[CONSTANT_STYLES.TXT_BASE, styles.txt]}>BACK</Text>
          </TouchableOpacity>
        )}
        <TouchableOpacity
          style={[CONSTANT_STYLES.BG_RED, styles.btn, { right: 25, bottom: 0 }]}
          onPress={props.onSubmit}
        >
          <Text allowFontScaling={false} style={[CONSTANT_STYLES.TXT_BASE, styles.txt]}>
            SUBMIT
          </Text>
        </TouchableOpacity>
      </View>
    </>
  );
};

const styles = StyleSheet.create({
  container: {
    flex: 1,
    flexDirection: "row",
    padding: 15,
    paddingTop: 25,
  },
  line: {
    height: 2,
    backgroundColor: "#707070",
    width: 50,
    alignSelf: "center",
    marginTop: -35,
    flexShrink: 1,
  },
  btn: {
    paddingLeft: 20,
    paddingRight: 20,
    paddingTop: 8,
    paddingBottom: 8,
    borderRadius: 10,
    position: "absolute",
  },
  txt: {
    fontWeight: "bold",
  },
});
