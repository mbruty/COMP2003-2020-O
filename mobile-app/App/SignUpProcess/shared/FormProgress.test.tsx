import React from "react";
import renderer from "react-test-renderer";
import { FormProgress } from "./FormProgress";

test("Renders first item in form progress", () => {
  const circleBtnList = renderer
    .create(<FormProgress selectedIdx={0} />)
    .toJSON();
  expect(circleBtnList).toMatchSnapshot();
});

test("Renders second item in form progress", () => {
  const circleBtnList = renderer
    .create(<FormProgress selectedIdx={1} />)
    .toJSON();
  expect(circleBtnList).toMatchSnapshot();
});

test("Renders third item in form progress", () => {
  const circleBtnList = renderer
    .create(<FormProgress selectedIdx={2} />)
    .toJSON();
  expect(circleBtnList).toMatchSnapshot();
});
