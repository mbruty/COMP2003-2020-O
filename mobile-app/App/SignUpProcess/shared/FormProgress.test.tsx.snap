// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders first item in form progress 1`] = `
Array [
  <View
    style={
      Array [
        Object {
          "flex": 1,
          "flexDirection": "row",
          "padding": 15,
          "paddingTop": 25,
        },
        Object {
          "alignSelf": "center",
          "justifyContent": "space-between",
          "width": 725,
        },
      ]
    }
  >
    <View
      style={
        Object {
          "alignItems": "center",
        }
      }
    >
      <Image
        source={1}
        style={
          Object {
            "height": 35,
            "width": 35,
          }
        }
      />
      <Text
        allowFontScaling={false}
        style={
          Object {
            "textAlign": "center",
          }
        }
      >
        Create Your 
Account
      </Text>
    </View>
    <View
      style={
        Object {
          "alignSelf": "center",
          "backgroundColor": "#707070",
          "flexShrink": 1,
          "height": 2,
          "marginTop": -35,
          "width": 50,
        }
      }
    />
    <View
      style={
        Object {
          "alignItems": "center",
        }
      }
    >
      <Image
        source={1}
        style={
          Object {
            "height": 35,
            "width": 35,
          }
        }
      />
      <Text
        allowFontScaling={false}
        style={
          Object {
            "textAlign": "center",
          }
        }
      >
        Confirm Your 
Email
      </Text>
    </View>
    <View
      style={
        Object {
          "alignSelf": "center",
          "backgroundColor": "#707070",
          "flexShrink": 1,
          "height": 2,
          "marginTop": -35,
          "width": 50,
        }
      }
    />
    <View
      style={
        Object {
          "alignItems": "center",
        }
      }
    >
      <Image
        source={1}
        style={
          Object {
            "height": 35,
            "width": 35,
          }
        }
      />
      <Text
        allowFontScaling={false}
        style={
          Object {
            "textAlign": "center",
          }
        }
      >
        Select Your 
Preferences
      </Text>
    </View>
  </View>,
  <View
    style={
      Array [
        Object {
          "flex": 1,
          "flexDirection": "row",
          "padding": 15,
          "paddingTop": 25,
        },
        Object {
          "paddingBottom": 25,
        },
      ]
    }
  >
    <View
      accessible={true}
      focusable={false}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "backgroundColor": "#FD4040",
          "borderRadius": 10,
          "bottom": 0,
          "opacity": 1,
          "paddingBottom": 8,
          "paddingLeft": 20,
          "paddingRight": 20,
          "paddingTop": 8,
          "position": "absolute",
          "right": 25,
        }
      }
    >
      <Text
        allowFontScaling={false}
        style={
          Array [
            Object {
              "color": "#F1F1F1",
            },
            Object {
              "fontWeight": "bold",
            },
          ]
        }
      >
        SUBMIT
      </Text>
    </View>
  </View>,
]
`;

exports[`Renders second item in form progress 1`] = `
Array [
  <View
    style={
      Array [
        Object {
          "flex": 1,
          "flexDirection": "row",
          "padding": 15,
          "paddingTop": 25,
        },
        Object {
          "alignSelf": "center",
          "justifyContent": "space-between",
          "width": 725,
        },
      ]
    }
  >
    <View
      style={
        Object {
          "alignItems": "center",
        }
      }
    >
      <Image
        source={1}
        style={
          Object {
            "height": 35,
            "width": 35,
          }
        }
      />
      <Text
        allowFontScaling={false}
        style={
          Object {
            "textAlign": "center",
          }
        }
      >
        Create Your 
Account
      </Text>
    </View>
    <View
      style={
        Object {
          "alignSelf": "center",
          "backgroundColor": "#707070",
          "flexShrink": 1,
          "height": 2,
          "marginTop": -35,
          "width": 50,
        }
      }
    />
    <View
      style={
        Object {
          "alignItems": "center",
        }
      }
    >
      <Image
        source={1}
        style={
          Object {
            "height": 35,
            "width": 35,
          }
        }
      />
      <Text
        allowFontScaling={false}
        style={
          Object {
            "textAlign": "center",
          }
        }
      >
        Confirm Your 
Email
      </Text>
    </View>
    <View
      style={
        Object {
          "alignSelf": "center",
          "backgroundColor": "#707070",
          "flexShrink": 1,
          "height": 2,
          "marginTop": -35,
          "width": 50,
        }
      }
    />
    <View
      style={
        Object {
          "alignItems": "center",
        }
      }
    >
      <Image
        source={1}
        style={
          Object {
            "height": 35,
            "width": 35,
          }
        }
      />
      <Text
        allowFontScaling={false}
        style={
          Object {
            "textAlign": "center",
          }
        }
      >
        Select Your 
Preferences
      </Text>
    </View>
  </View>,
  <View
    style={
      Array [
        Object {
          "flex": 1,
          "flexDirection": "row",
          "padding": 15,
          "paddingTop": 25,
        },
        Object {
          "paddingBottom": 25,
        },
      ]
    }
  >
    <View
      accessible={true}
      focusable={false}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "backgroundColor": "#FD4040",
          "borderRadius": 10,
          "bottom": 0,
          "opacity": 1,
          "paddingBottom": 8,
          "paddingLeft": 20,
          "paddingRight": 20,
          "paddingTop": 8,
          "position": "absolute",
          "right": 25,
        }
      }
    >
      <Text
        allowFontScaling={false}
        style={
          Array [
            Object {
              "color": "#F1F1F1",
            },
            Object {
              "fontWeight": "bold",
            },
          ]
        }
      >
        SUBMIT
      </Text>
    </View>
  </View>,
]
`;

exports[`Renders third item in form progress 1`] = `
Array [
  <View
    style={
      Array [
        Object {
          "flex": 1,
          "flexDirection": "row",
          "padding": 15,
          "paddingTop": 25,
        },
        Object {
          "alignSelf": "center",
          "justifyContent": "space-between",
          "width": 725,
        },
      ]
    }
  >
    <View
      style={
        Object {
          "alignItems": "center",
        }
      }
    >
      <Image
        source={1}
        style={
          Object {
            "height": 35,
            "width": 35,
          }
        }
      />
      <Text
        allowFontScaling={false}
        style={
          Object {
            "textAlign": "center",
          }
        }
      >
        Create Your 
Account
      </Text>
    </View>
    <View
      style={
        Object {
          "alignSelf": "center",
          "backgroundColor": "#707070",
          "flexShrink": 1,
          "height": 2,
          "marginTop": -35,
          "width": 50,
        }
      }
    />
    <View
      style={
        Object {
          "alignItems": "center",
        }
      }
    >
      <Image
        source={1}
        style={
          Object {
            "height": 35,
            "width": 35,
          }
        }
      />
      <Text
        allowFontScaling={false}
        style={
          Object {
            "textAlign": "center",
          }
        }
      >
        Confirm Your 
Email
      </Text>
    </View>
    <View
      style={
        Object {
          "alignSelf": "center",
          "backgroundColor": "#707070",
          "flexShrink": 1,
          "height": 2,
          "marginTop": -35,
          "width": 50,
        }
      }
    />
    <View
      style={
        Object {
          "alignItems": "center",
        }
      }
    >
      <Image
        source={1}
        style={
          Object {
            "height": 35,
            "width": 35,
          }
        }
      />
      <Text
        allowFontScaling={false}
        style={
          Object {
            "textAlign": "center",
          }
        }
      >
        Select Your 
Preferences
      </Text>
    </View>
  </View>,
  <View
    style={
      Array [
        Object {
          "flex": 1,
          "flexDirection": "row",
          "padding": 15,
          "paddingTop": 25,
        },
        Object {
          "paddingBottom": 25,
        },
      ]
    }
  >
    <View
      accessible={true}
      focusable={false}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "backgroundColor": "#FD4040",
          "borderRadius": 10,
          "bottom": 0,
          "opacity": 1,
          "paddingBottom": 8,
          "paddingLeft": 20,
          "paddingRight": 20,
          "paddingTop": 8,
          "position": "absolute",
          "right": 25,
        }
      }
    >
      <Text
        allowFontScaling={false}
        style={
          Array [
            Object {
              "color": "#F1F1F1",
            },
            Object {
              "fontWeight": "bold",
            },
          ]
        }
      >
        SUBMIT
      </Text>
    </View>
  </View>,
]
`;
